diff --git a/dist/client/components/router-reducer/get-prefetch-cache-entry-status.js b/dist/client/components/router-reducer/get-prefetch-cache-entry-status.js
index b611280387bf47e37f797483ef93ae6e87755939..d2f4f31300c6539a4306dc1e8e432972cf37d43a 100644
--- a/dist/client/components/router-reducer/get-prefetch-cache-entry-status.js
+++ b/dist/client/components/router-reducer/get-prefetch-cache-entry-status.js
@@ -30,6 +30,7 @@ var PrefetchCacheEntryStatus;
     PrefetchCacheEntryStatus["stale"] = "stale";
 })(PrefetchCacheEntryStatus || (PrefetchCacheEntryStatus = {}));
 function getPrefetchEntryCacheStatus(param) {
+    return "stale";
     let { kind , prefetchTime , lastUsedTime  } = param;
     // if the cache entry was prefetched or read less than 30s ago, then we want to re-use it
     if (Date.now() < (lastUsedTime != null ? lastUsedTime : prefetchTime) + THIRTY_SECONDS) {
diff --git a/dist/esm/client/components/router-reducer/get-prefetch-cache-entry-status.js b/dist/esm/client/components/router-reducer/get-prefetch-cache-entry-status.js
index 7156f75d5fd9115724366eecf1a21cf48f3f0b9a..f6627486df2714143ca00fd2fca563a4147d4034 100644
--- a/dist/esm/client/components/router-reducer/get-prefetch-cache-entry-status.js
+++ b/dist/esm/client/components/router-reducer/get-prefetch-cache-entry-status.js
@@ -8,6 +8,7 @@ export var PrefetchCacheEntryStatus;
     PrefetchCacheEntryStatus["stale"] = "stale";
 })(PrefetchCacheEntryStatus || (PrefetchCacheEntryStatus = {}));
 export function getPrefetchEntryCacheStatus(param) {
+    return "stale";
     let { kind , prefetchTime , lastUsedTime  } = param;
     // if the cache entry was prefetched or read less than 30s ago, then we want to re-use it
     if (Date.now() < (lastUsedTime != null ? lastUsedTime : prefetchTime) + THIRTY_SECONDS) {